### PR TITLE
Use Roe-averaged state in HLLE to compute wave speeds.

### PR DESCRIPTION
**Pull request by forrestglines**
Changes HLLE to compute wave speeds from the L/R/Roe-averaged states like Athena++ does for adiabatic hydro.

Currently this only changes HLLE for hydro. I'm still exploring whether this change affects disks in a private branch for curvilinear coordinates.

### CODE REVIEW

**Comment by pgrete** [comment]
Any results on the curvilinear coords (or in other words, should we go ahead and merge this)?

**Comment by forrestglines** [comment]
@par-hermes format

**Comment by forrestglines** [comment]
> Any results on the curvilinear coords (or in other words, should we go ahead and merge this)?

We should go ahead and merge this. Although I don't have any comparisons on hand for Cartesian (it passes tests at least), these changes improved the stability of stratified disks in cylindrical and spherical coordinates.

Changing the hydro HLLE also won't affect the INCITE runs since for MHD we use HLLD and hydro we use HLLC.

File: src/hydro/rsolvers/hydro_hlle.hpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ struct Riemann<Fluid::euler, RiemannSolver::hlle> {
     Real gm1 = gamma - 1.0;
     Real igm1 = 1.0 / gm1;
     parthenon::par_for_inner(member, il, iu, [&](const int i) {
-      Real wli[(NHYDRO)], wri[(NHYDRO)];
+      Real wli[(NHYDRO)], wri[(NHYDRO)], wroe[(NHYDRO)];
```
Comment (pgrete) [inline src/hydro/rsolvers/hydro_hlle.hpp:53]: We're only using three out of five variable in `wroe[(NHYDRO)]` so from a performance point of view (kernel size) might be worth to just use three plain variables.

**Review by pgrete — APPROVED** [review]
LGTM

File: src/hydro/rsolvers/hydro_hlle.hpp (hunk 1 above)
Comment (forrestglines) [reply src/hydro/rsolvers/hydro_hlle.hpp:53]: I'll make a note of this for the flux optimizations I'm doing. Register pressure might be a big bottle neck on the flux kernel performance. This fix would be useful, although I suspect the compile is already optimizing it out.

**Pull request merged**